### ISSUE TITLE
Is Style/FileOpen applicable for class/global variables?

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#14997]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Style/FileOpen

### Steps to reproduce the problem

1. How is ...

```ruby
def open
  File.open('foo')
end

def bar(str)
  $baz ||= open
  $baz.write(str)
end
```

... better than ...

```ruby
def bar(str)
  $baz ||= File.open('foo') # Style/FileOpen
  $baz.write(str)
end
```

2. And ....

```ruby
class Foo
  def initialize
    @file = open
  end

  def open
    File.open('foo')
  end

  def write(str)
    @file.write(str)
  end
end
```

... better than ...

```ruby
class Foo
  def initialize
    @file = File.open('foo') # Style/FileOpen
  end

  def write(str)
    @file.write(str)
  end
end
```

### Expected behavior

Does the cop actually serve its purpose (preventing fd leaks) in these cases, or just add verbosity?

### Actual behavior

Style/FileOpen

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
System Version: macOS 26.3 (25D125)
Model Identifier: MacBookAir10,1
ruby 4.0.1 (2026-01-13 revision e04267a14b) +PRISM [arm64-darwin25]
rubocop     2026-03-06 b7d518bb6
```
